galasa-extensions-parent/dev.galasa.ecosystem.docker.builder/src/main/java/dev/galasa/ecosystem/docker/builder/DockerEcosystemBuilder.java

```diff
@@ -0,0 +1,525 @@
+package dev.galasa.ecosystem.docker.builder;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.Map;
+import java.util.Scanner;
+
+import org.yaml.snakeyaml.Yaml;
+
+public class DockerEcosystemBuilder {
+
+	public static void main(String[] args) throws FileNotFoundException {
+		
+		Scanner scanner = new Scanner(System.in);
+		System.out.println("Welcome to the Galasa Ecosystem manager for Docker \n");
+		System.out.println("If you have not run this program before, default values for the configuration have been supplied \n");
+		System.out.println("Here is your current configuration for the Ecosystem: \n");
+		String configLocation = "classes/";
+		
+		try {
+			BufferedReader dis = new BufferedReader(new FileReader(configLocation + "config.yml"));
+			String line;
+			while((line = dis.readLine()) != null)
+			{
+			    System.out.println(line);
+			}
+			dis.close();
+			
+		} catch (FileNotFoundException e) {
+			System.out.println("File not found");
+			e.printStackTrace();
+		} catch (IOException e) {
+
+			e.printStackTrace();
+		}
+		
+		System.out.println("\n" + "Would you like to alter any of these properties? (Y or N, Default Y)");
+		String reuse = scanner.nextLine();
+		while (!(reuse.toUpperCase().equals("Y") || reuse.toUpperCase().equals("N") || reuse.isEmpty())) {	
+			System.out.println("Please enter Y or N");
+			reuse = scanner.nextLine();
+		}
+		if (reuse.toUpperCase().equals("Y") || reuse.isEmpty()) {			
+			alterProps();
+		
+		}
+		else if (reuse.toUpperCase().equals("N")){
+			System.out.println("Generating script..." + "\n");
+			try {
+				generateScript();
+			} catch (FileNotFoundException e) {
+				System.out.println("Error");
+				e.printStackTrace();
+			} catch (UnsupportedEncodingException e) {
+				System.out.println("Error");
+				e.printStackTrace();
+			}
+		}
+	    scanner.close();
+		
+	}
+	
+	static void alterProps() throws FileNotFoundException {
+		System.out.println("\n" + "Which of these properties would you like to change? Please enter the property name(s) from above you wish to change all one line with spaces between each");
+		System.out.println("\n" + "Enter exit to return to the main screen");
+		Scanner scanner = new Scanner(System.in);
+		String propList = scanner.nextLine();
+		while (propList.isEmpty()) {
+			System.out.println("Please enter which properties you would like to edit" + "\n");
+			alterProps();
+		}
+		if (propList.toUpperCase().contains("EXIT")) {
+			System.out.println("Returning you to the main screen...");
+			main(null);
+		}
+		if (propList.toUpperCase().contains("HOSTNAME")) {
```

Comment:
Each of these options should be a individual method for readability.     The method is called via a switch/case statement